bigtop-deploy/puppet/modules/spark/templates/spark-defaults.conf

```diff
@@ -13,7 +13,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+<% if @master_url -%>
 spark.master <%= @master_url %>
+<% else -%>
+<% if scope['deploy::roles'].include? 'spark-on-yarn' -%>
+spark.master yarn
+<% else -%>
+spark.master spark://<%= @master_host %>:<%= @master_port %>
```

Comment:
This block generally LGTM, but note that this is effectively making `spark://host:port` the default master.  Prior to this change, people deploying the `spark` component would get `yarn` as the master by default.

So I'm +1 here, but it may be worth mentioning on-list or in the release notes that the default spark configuration is changing.